cli_tools/gce_vm_image_import/main.go

```diff
@@ -300,37 +209,7 @@ func buildDaisyVars(translateWorkflowPath string) map[string]string {
 	return varMap
 }
 
-func createStorageClient(ctx context.Context) *storage.Client {
-	storageOptions := []option.ClientOption{option.WithCredentialsFile(*oauth)}
-	storageClient, err := storage.NewClient(ctx, storageOptions...)
-	if err != nil {
-		log.Fatalf("error creating storage client %v", err)
-	}
-	return storageClient
-}
-
-// creates a new Daisy Compute client
-func createComputeClient(ctx *context.Context) daisycompute.Client {
-	computeOptions := []option.ClientOption{option.WithCredentialsFile(*oauth)}
-	if *ce != "" {
-		computeOptions = append(computeOptions, option.WithEndpoint(*ce))
-	}
-
-	computeClient, err := daisycompute.NewClient(*ctx, computeOptions...)
-	if err != nil {
-		log.Fatalf("compute client: %v", err)
-	}
-	return computeClient
-}
-
-func runImport(ctx context.Context, metadataGCEHolder computeutils.MetadataGCE,
-	scratchBucketCreator *gcevmimageimportutil.ScratchBucketCreator,
-	zoneRetriever *gcevmimageimportutil.ZoneRetriever, storageClient commondomain.StorageClientInterface) error {
-
-	err := populateMissingParameters(&metadataGCEHolder, scratchBucketCreator, zoneRetriever, storageClient)
-	if err != nil {
-		return err
-	}
+func runImport(ctx context.Context) error {
 	importWorkflowPath, translateWorkflowPath := getWorkflowPaths()
 	varMap := buildDaisyVars(translateWorkflowPath)
 	workflow, err := daisyutils.ParseWorkflow(&computeutils.MetadataGCE{}, importWorkflowPath, varMap,
```

Comment:
I read into why metadatagce is still needed after param population, and see it's actually doing dup thing (populate project/zone) in daisyutils.ParseWorkflow. So I removed that dup logic from daisyutils.ParseWorkflow due that all callers have handled it. Removed UT as well because corresponding UT exists in param_utils_test

Reply:
What I meant is you create a new MetadataGCE. It's not a big deal as this is just a struct to enabled polymorphism.